src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
 
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
 
-}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0
+    || this.b <= 0
+    || this.c <= 0
+    || this.a >= this.b + this.c
+    || this.b >= this.a + this.c
+    || this.c >= this.a + this.b
+    ) {
+      throw new Error('Wrong input data');
```

Comment:
Checklist p3
How should the user understand what was wrong with his values?
Be more detailed, and write what data are invalid.

Reply:
p.s. In all errors

Reply:
Got it, ty